### DISCUSSION TITLE
High order polynomial libraries with Weak SINDy

### DISCUSSION THREAD

**Discussion by scatr** [dynamicslab/pysindy#429] [General]
Hi, 

Is it possible to use the `ps.PolynomialLibrary` with the `ps.WeakPDELibrary`. I'd like to build a library with high-order polynomials (up to order 7). As far as I can see, we have specify the functions as lists for `ps.WeakPDELibrary` which becomes a bit of a pain for high order polynomial libraries. 

Thanks
Alasdair

**Comment by Jacob-Stevens-Haas** [comment]
An excellent argument in favor of #351 :).  The short answer is no, the long answer is I'd be happy for a colleague to work on this.  It's not super relevant to my research, but what I believe needs to happen is that `WeakPDELibrary` would need to become a `WeakSINDy` class (easier), and we need to formalize the abstractions involved in calculating no-derivative and maybe mixed derivative/non-derivative terms (e.g. $u\cdot u_x$) (harder).  One could start by extracting some functionality from around [here](https://github.com/dynamicslab/pysindy/blob/2633ee091d5648cd67bb061f5888e44adbf14a5c/pysindy/feature_library/weak_pde_library.py#L878) in `WeakPDELibrary`.

**Comment by znicolaou** [comment]
FYI @scatr & @Jacob-Stevens-Haas , I put a little thought into this, and started a branch and draft pull request #433 to try out a solution here. If you have thoughts, please let me know!
